libsoftwarecontainer/unit-test/networkgatewayfunctions_unittest.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2017 Pelagicore AB
+ *
+ * Permission to use, copy, modify, and/or distribute this software for
+ * any purpose with or without fee is hereby granted, provided that the
+ * above copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+ * WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+ * BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+ * OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+ * WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+ * ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+ * SOFTWARE.
+ *
+ * For further information see LICENSE
+ */
+
+
+#include "gateway/network/networkgatewayfunctions.h"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+using namespace softwarecontainer;
+
+struct testParams
+{
+    uint32_t    netmask;
+    std::string gatewayIP;
+    int32_t     containerID;
+    bool        expectedReturn;
+};
+
+class NetworkGatewayFunctionsTests : public  ::testing::TestWithParam<testParams>
+{
+public:
+    NetworkGatewayFunctions ngf;
+    testParams testparams;
+
+    void SetUp() override
+    {
+        testparams = GetParam();
+    }
+};
+
+/*
+ * This data is fed to the NetworkGatewayFunctions tests
+ */
+INSTANTIATE_TEST_CASE_P(NetworkFunctionsParameters, NetworkGatewayFunctionsTests, ::testing::Values(
+        testParams{32, "192.168.0.0", 0, false},
+
+        testParams{31, "192.168.0.0", 0, true},
+        testParams{31, "192.168.0.0", 1, false},
+
+        testParams{30, "192.168.0.0", 2, true},
+        testParams{30, "192.168.0.0", 3, false},
+        testParams{30, "192.168.0.0", 4, false},
+
+        testParams{29, "192.168.0.0", 6, true},
+        testParams{29, "192.168.0.0", 7, false},
+        testParams{29, "192.168.0.0", 8, false},
+
+        testParams{28, "192.168.0.0", 14, true},
+        testParams{28, "192.168.0.0", 15, false},
+        testParams{28, "192.168.0.0", 16, false},
+
+        testParams{27, "192.168.0.0", 30, true},
+        testParams{27, "192.168.0.0", 31, false},
+        testParams{27, "192.168.0.0", 32, false},
+
+        testParams{16, "192.168.0.0", 1024, true},
+        testParams{16, "192.168.0.0", 65534, true},
+        testParams{16, "192.168.0.0", 65535, false},
+        testParams{16, "192.168.0.0", 65536, false},
+
+        testParams{24, "192.168.0.0", 0, true},
+        testParams{24, "192.168.0.0", 1, true},
+        testParams{24, "192.168.0.0", 10, true},
+        testParams{24, "192.168.0.0", 100, true},
+        testParams{24, "192.168.0.0", 255, false}
+));
```

Comment:
Nice use of parametrization! I think it would be good with inline comments on why certain things are expected to fail etc. though.